### PR TITLE
fix(filters): use Date() parsing for dates

### PR DESCRIPTION
**Pull request by jniles**
This commit removes the moment parsing and just converts dates to JS dates and then lets mysqljs deal with them.

### CODE REVIEW

**Comment by jniles** [comment]
So, in practice, this accomplishes the same thing.  However, it is faster.  You can test by putting the following in a `test.js` file and running `node test.js` as I did:
```js
const moment = require('moment');                                                                                                                                                                                  
                                                                                                                                                                                                                   
const ROUNDS = 100000;                                                                                                                                                                                             
const date = '2018-05-31T23:00:00.000Z';                                                                                                                                                                           
                                                                                                                                                                                                                   
let start = Date.now();                                                                                                                                                                                            
let i = 0;                                                                                                                                                                                                         
let target;                                                                                                                                                                                                        
                                                                                                                                                                                                                   
while (i < ROUNDS) {                                                                                                                                                                                               
  i++;                                                                                                                                                                                                             
  target = new Date(date);                                                                                                                                                                                         
}                                                                                                                                                                                                                  
                                                                                                                                                                                                                   
let end = Date.now();                                                                                                                                                                                              
                                                                                                                                                                                                                   
console.log('new Date(date)');                                                                                                                                                                                     
console.log('start:', start);                                                                                                                                                                                      
console.log('end:', end);                                                                                                                                                                                          
console.log('elapsed:', end - start);                                                                                                                                                                              
                                                                                                                                                                                                                   
                                                                                                                                                                                                                   
start = Date.now();                                                                                                                                                                                                
i = 0;                                                                                                                                                                                                             
target = null;                                                                                                                                                                                                     
                                                                                                                                                                                                                   
while (i < ROUNDS) {                                                                                                                                                                                               
  i++;                                                                                                                                                                                                             
  target = moment(date).format('YYYY-MM-DD').toString();                                                                                                                                                           
}                                                                                                                                                                                                                  
                                                                                                                                                                                                                   
end = Date.now();                                                                                                                                                                                                  
                                                                                                                                                                                                                   
console.log('moment(date)');                                                                                                                                                                                       
console.log('start:', start);                                                                                                                                                                                      
console.log('end:', end);                                                                                                                                                                                          
console.log('elapsed:', end - start); 
```

My results:
```
new Date(date)
start: 1529914946749
end: 1529914946791
elapsed: 42

moment(date)
start: 1529914946794
end: 1529914948759
elapsed: 1965
```

File: server/lib/filter.js [hunk 1]
```diff
@@ -117,11 +107,11 @@ class FilterParser {
    */
   dateFrom(filterKey, columnAlias = filterKey, tableAlias = this._tableAlias) {
     const tableString = this._formatTableAlias(tableAlias);
+    const timestamp = this._filters[filterKey];
```
Comment (mbayopanda) [inline server/lib/filter.js:110]: :+1

**Review by mbayopanda — APPROVED** [review]

**Comment by mbayopanda** [comment]
I test it locally it seems working perfectly.

bors r+

**Comment by jniles** [comment]
bors r+

One last try...

**Pull request merged**